BrowserKit/Sources/ToolbarKit/AddressToolbar/BrowserAddressToolbar.swift

```diff
@@ -50,16 +52,27 @@ public class BrowserAddressToolbar: UIView, AddressToolbar, ThemeApplicable, Loc
     private var toolbarBottomBorderHeightConstraint: NSLayoutConstraint?
     private var leadingNavigationActionStackConstraint: NSLayoutConstraint?
     private var trailingBrowserActionStackConstraint: NSLayoutConstraint?
+    private var locationContainerHeightConstraint: NSLayoutConstraint?
 
     override init(frame: CGRect) {
         super.init(frame: .zero)
         setupLayout()
+        setupObservers()
+        adjustHeightConstraintForCurrentContentSizeCategory()
     }
 
     required init?(coder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
 
+    deinit {
+        NotificationCenter.default.removeObserver(
```

Comment:
Nit: This doesn't need to be removed anymore - iOS is doing this automagically on deinit.